transformations/chinese_antonym_synonym_substitution/transformation.py

```diff
@@ -0,0 +1,107 @@
+import itertools
+import random
+import os
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+import jieba
+from nlpcda import Similarword
+
+def chinese_antonym_synonym_substitution(text,
+                          prob,
+                          seed,
+                          max_outputs,
+                          config,
+                          chinese_antonym_data
+                          ):
+    random.seed(seed)
+
+    perturbed_texts = []
+    word1_list = [chinese_antonym['word1'] for chinese_antonym in chinese_antonym_data]
+    word2_list = [chinese_antonym['word2'] for chinese_antonym in chinese_antonym_data]
+
+    if(config == 'synonym'):
+        smw = Similarword(create_num=max_outputs+1, change_rate=prob)
+        output_list = smw.replace(text)
+
+        # Notice that the first input is the non-perturb version
+        output_list = output_list[1:]
+        perturbed_texts = output_list
+
+    if(config == 'antonym'):
+
+        output = jieba.lcut(text)
+        for _ in itertools.repeat(None, max_outputs):
+            butter_text = ""
```

Comment:
You should use a list to accumulate the words before joining into a string instead of appending, which creates a new string every time.